nilearn/glm/first_level/first_level.py

```diff
@@ -800,13 +807,31 @@ def _get_voxelwise_model_attribute(self, attribute,
         return output
 
 
+def _check_repetition_time(t_r):
+    if not isinstance(t_r, (float, int)):
+        raise TypeError("'t_r' must be a float or an integer. "
+                        f"Got {type(t_r)} instead.")   
+    if t_r <= 0:
+        raise ValueError("'t_r' must be positive. "
+                        f"Got {t_r} instead.")        
+
+
+def _check_slice_time_ref(slice_time_ref):
+    if not isinstance(slice_time_ref, (float, int)):
+        raise TypeError("'slice_time_ref' must be a float or an integer. "
+                        f"Got {type(slice_time_ref)} instead.")   
+    if slice_time_ref < 0 or slice_time_ref > 1:
+        raise ValueError("'slice_time_ref' must be between 0 and 1. "
+                        f"Got {slice_time_ref} instead.")  
+
+
 def first_level_from_bids(dataset_path,
                           task_label,
                           space_label=None,
                           sub_labels=None,
                           img_filters=None,
                           t_r=None,
-                          slice_time_ref=0.,
+                          slice_time_ref=None,
```

Comment:
Amigos, we need a deprecation warning before changing the API !

Reply:
I actually need to maybe rethink whether the change of API is even desirable / necessary.